Sources/Sentry/SentryDependencyContainer.m

```diff
@@ -172,16 +174,29 @@ - (instancetype)init
 - (SentryFileManager *)fileManager SENTRY_THREAD_SANITIZER_DOUBLE_CHECKED_LOCK
 {
     SENTRY_LAZY_INIT(_fileManager, ({
-        NSError *error;
-        SentryFileManager *manager = [[SentryFileManager alloc] initWithOptions:SentrySDK.options
-                                                                          error:&error];
-        if (manager == nil) {
-            SENTRY_LOG_DEBUG(@"Could not create file manager - %@", error);
+        SentryOptions *options = SentrySDK.options;
+        if (options == nil) {
+            SENTRY_LOG_FATAL(
+                @"SentryDependencyContainer.fileManager called before SentrySDK.options was set.");
         }
-        manager;
+        [self getFileManagerForOptions:options];
     }));
 }
 
+- (SentryFileManager *_Nullable)getFileManagerForOptions:(SentryOptions *)options
+    SENTRY_THREAD_SANITIZER_DOUBLE_CHECKED_LOCK
+{
+    NSError *error;
+    SentryFileManager *manager =
+        [[SentryFileManager alloc] initWithOptions:options
+                              dispatchQueueWrapper:self.dispatchQueueWrapper
+                                             error:&error];
+    if (manager == nil) {
+        SENTRY_LOG_FATAL(@"Could not create file manager - %@", error);
+    }
+    return manager;
```

Comment:
`h`: The way the synchronization currently works in the SentryFileManager, we must ensure only having one SentryFileManager instance per DSN, as you already pointed out in the PR description. Otherwise, we end up with race conditions when accessing the files.
I think your idea of keeping weak references using the DSN with the key and the file manager as a value could work.